ui/pages/extension/[...component].js

```diff
@@ -48,53 +49,99 @@ function matchComponentURI(extensionURI, currentURI) {
 }
 
 
-class RemoteExtension extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      componentTitle : '',
-      isLoading : true,
-      capabilitiesRegistryObj : null,
+const RemoteExtension=(props) => {
+  // constructor(props) {
+  //   super(props);
+  //   this.state = {
+  //     componentTitle : '',
+  //     isLoading : true,
+  //     capabilitiesRegistryObj : null,
+  //   }
+  // }
+  const [componentTitle, setComponentTitle] = useState('');
+  const [isLoading, setIsLoading] = useState(true);
+  const [capabilitiesRegistryObj, setCapabilitiesRegistryObj] = useState(null);
+  const router = useRouter();
+  const prevProps = useRef();
+
+  // componentWillUnmount() {
+  //   this.setState({
+  //     componentTitle : '',
+  //     isLoading : true,
+  //     capabilitiesRegistryObj : null,
+  //   })
+  // }
+  useEffect(() => {
+    prevProps.current = props;``
```

Comment:
what is the use of this `useEffect`

Reply:
well I put useEffect there to keep track of props that can change between renders

Reply:
no please ,  you dont need useEffect to track props they automatically do .